patterns/direct_processor.py

```diff
@@ -0,0 +1,142 @@
+
+# Goal is to demonstrate receive-process-reply (processor) pattern. Processor subscribes to a request topic and upon receipt of 
+# request message carrying a string, the processor converts the payload to uppercase (for simplicity) and publishes to a 
+# designated topic.
+#   direct_requestor: 
+#       subscribes to request topic 'solace/samples/python/direct/processor/input/{count}
+#       proceses the payload
+#       publishes to reply topic 'solace/samples/python/direct/processor/output'
+
+TOPIC_PREFIX = "solace/samples/python"
+
+# Broker Config. Note: Could pass other properties Look into
+broker_props = {
+    "solace.messaging.transport.host": os.environ.get('SOLACE_HOST') or "tcp://localhost:55555,tcp://localhost:55554",
+    "solace.messaging.service.vpn-name": os.environ.get('SOLACE_VPN') or "default",
+    "solace.messaging.authentication.scheme.basic.username": os.environ.get('SOLACE_USERNAME') or "default",
+    "solace.messaging.authentication.scheme.basic.password": os.environ.get('SOLACE_PASSWORD') or ""
+    }
+
+# Build A messaging service with a reconnection strategy of 20 retries over an interval of 3 seconds
+# Note: The reconnections strategy could also be configured using the broker properties object
+messaging_service = MessagingService.builder().from_properties(broker_props)\
+                        .with_reconnection_retry_strategy(RetryStrategy.parametrized_retry(20,3))\
+                        .build()
+
+# Blocking connect thread
+messaging_service.connect()
+print(f'\nMessaging Service connected? {messaging_service.is_connected}')
+
+# Inner classes for connection error handling
+class ServiceEventHandler(ReconnectionListener, ReconnectionAttemptListener, ServiceInterruptionListener):
+    def on_reconnected(self, e: ServiceEvent):
+        print("\non_reconnected")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+    
+    def on_reconnecting(self, e: "ServiceEvent"):
+        print("\non_reconnecting")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+    def on_service_interrupted(self, e: "ServiceEvent"):
+        print("\non_service_interrupted")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+# Event Handling for the messaging service
+service_handler = ServiceEventHandler()
+messaging_service.add_reconnection_listener(service_handler)
+messaging_service.add_reconnection_attempt_listener(service_handler)
+messaging_service.add_service_interruption_listener(service_handler)
+ 
+# Inner classes for publish error handling
+class PublisherErrorHandling(PublishFailureListener):
+    def on_failed_publish(self, e: "FailedPublishEvent"):
+        print("on_failed_publish")
+
+# Handle received messages
+class MessageHandlerImpl(MessageHandler):
+    def on_message(self, message: InboundMessage):
+        out_topic = Topic.of(TOPIC_PREFIX + f'/direct/processor/output')
```

Comment:
can we change var name to `publish_topic`

Reply:
Done - also changed the in_topic -> subscribe_topic